lib/validates_zipcode/zipcode.rb

```diff
@@ -7,24 +7,33 @@ def initialize(args = {})
       @zipcode                = args.fetch(:zipcode)
       @country_alpha2         = args.fetch(:country_alpha2)
       @excluded_country_codes = args.fetch(:excluded_country_codes, [])
+      @format                 = args.fetch(:format, false)
     end
 
     def valid?
       return true if @excluded_country_codes.include?(@country_alpha2)
       return true unless regexp
-      !!(regexp =~ @zipcode)
+
+      zipcode = @format ? formatted_zip : @zipcode
+      !!(regexp =~ zipcode)
     end
     alias_method :validate, :valid?
 
     def format
-      raise InvalidZipcodeError, "invalid zipcode #{@zipcode} for country #{@country_alpha2.to_s.upcase}" unless valid?
-      Formatter.new(zipcode: @zipcode, country_alpha2: @country_alpha2).format
+      @format = true
+      raise InvalidZipcodeError, "invalid zipcode #{formatted_zip} for country #{@country_alpha2.to_s.upcase}" unless valid?
```

Comment:
This is confusing because it will always use formatted_zip even if `@format` is false.